### PR TITLE
Add Envoy filter types so typed_config can be used when configuring EnvoyFilters

### PR DESCRIPTION
**Pull request by nacx**
Import the Envoy filter packages to make sure the `init` method for each of them is called and the proto messages they define are properly registered.

This is needed when unmarshaling the `EnvoyFilter` patches, as the configuration objects may contain `typed_config` that needs to be properly deserialized. Without this patch an `unknown type: <type>` appears when configuring EnvoyFilters that define the `typed_config` property.

### CODE REVIEW

File: pkg/config/xds/filter_types.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+// Import all Envoy filter types so they are registered and deserialization does not fail
+// when using them in the "typed_config" attributes.
```
Comment (rshriram) [inline pkg/config/xds/filter_types.go]: Ah. So this is why people got an error when they used typed_config in their envoy filters! . Thanks for the fix..

**Comment by kyessenov** [comment]
We should be using v3 versions now. Also, just use UDPA struct instead of typed proto, and perhaps add a flag to load descriptors on startup.

**Comment by PiotrSikora** [comment]
I believe the plan is to move to v3 mid-year, so probably in 1.6 or 1.7.

Also, we don't have v3 merged in Istio Proxy just yet. Should be merged later today or tomorrow, though.

**Comment by kyessenov** [comment]
There'll be a small in CPU cost of proto roundtripping in envoy if we don't switch to v3. But it's not urgent.

**Comment by howardjohn** [comment]
do we have numbers on "small"?

**Pull request merged**